crates/matrix-sdk-crypto/src/machine/mod.rs

```diff
@@ -1336,7 +1359,54 @@ impl OlmMachine {
             e => self.handle_to_device_event(changes, &e).await,
         }
 
-        raw_event
+        Some(raw_event)
+    }
+
+    async fn event_is_from_dehydrated_device(
+        &self,
+        decrypted: &OlmDecryptionInfo,
+        sender_user_id: &UserId,
+    ) -> bool {
```

Comment:
it might come as a bit of a surprise that this accepts an `OlmDecryptionInfo` rather than, say, an `AnyToDeviceEvent`. Might be worth adding a doc-comment that explains that it works on *decrypted* to-device events.

Reply:
Done in 38ab50ecc2